### PR TITLE
dotnet-api-starter: add avatar upload

### PR DESCRIPTION
**Pull request by amonich**

### CODE REVIEW

File: services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs [hunk 1]
```diff
@@ -64,8 +66,50 @@ public async Task<IActionResult> UpdateCurrentAsync([FromBody] UpdateCurrentUser
...
+            });
+        }
+
+        [HttpPost("upload-photo")]
+        public async Task<IActionResult> UploadCurrentPhoto([FromForm] UploadPhotoModel model)
+        {
+            await using var stream = model.File.OpenReadStream();
+
+            var avatarUrl = await _cloudStorageService.UploadPublicAsync(
+                $"avatars/{CurrentUserId}-{DateTime.UtcNow:s}-{model.File.FileName}",
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs:80]: I think, there's no need to put datetime into file name. If we need uniqueness, it'd be better to generate id (but there's no need for it). So user id + file name should be enough.

File: services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs [hunk 2]
```diff
@@ -64,8 +66,50 @@ public async Task<IActionResult> UpdateCurrentAsync([FromBody] UpdateCurrentUser
...
+                stream
+            );
+
+            await _userService.UpdateAvatarUrlAsync(CurrentUserId, avatarUrl);
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs:84]: 1. To me it seems like this method should be responsible both for avatar upload and user update, especially considering ICloudStorageService is an infrastructure service and can be injected into IUserService.
2. As a result, more generic name (like UpdateAvatarAsync or SetAvatarAsync) should fit more.

File: services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs [hunk 3]
```diff
@@ -64,8 +66,50 @@ public async Task<IActionResult> UpdateCurrentAsync([FromBody] UpdateCurrentUser
...
+                user.FirstName,
+                user.LastName,
+                user.Email,
+                user.IsEmailVerified,
+                user.AvatarUrl
+            });
+        }
+
+        [HttpDelete("remove-photo")]
+        public async Task<IActionResult> RemoveCurrentPhoto()
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs:99]: `RemoveAvatarAsync`.

File: services/api-dotnet/src/app/Api.NoSql/appsettings.Development.json [hunk 4]
```diff
@@ -24,5 +24,12 @@
     "ConnectionString": "redis:6379", // Docker
     "AbsoluteExpirationInSeconds": 600,
     "SlidingExpirationInSeconds": 60
+  },
+  "CloudStorage": {
+    "Endpoint": "",
+    "BucketName": "",
+    "FolderName": "",
+    "AccessKey": "",
+    "Secret": ""
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Api.NoSql/appsettings.Development.json:33]: Let's move the whole section up into appsettings.json for consistency. And also replace empty strings with nulls (there's no particular difference - just for consistency).

File: services/api-dotnet/src/app/Common/Models/View/User/UserViewModel.cs [hunk 5]
```diff
@@ -9,4 +9,5 @@ public class UserViewModel
     public string FirstName { get; set; }
     public string LastName { get; set; }
     public string Email { get; set; }
+    public  string  AvatarUrl { get; set; }
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Common/Models/View/User/UserViewModel.cs:12]: Extra spaces.

File: services/api-dotnet/src/app/Common/Validators/User/UploadPhotoModelValidator.cs [hunk 6]
```diff
@@ -0,0 +1,23 @@
+﻿using Common.Models.View.User;
+using FluentValidation;
+
+namespace Common.Validators.User;
+
+public class UploadPhotoModelValidator : AbstractValidator<UploadPhotoModel>
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Common/Validators/User/UploadPhotoModelValidator.cs:6]: `UploadAvatarModelValidator`.

File: services/api-dotnet/src/app/Common/Common.csproj [hunk 7]
```diff
@@ -8,12 +8,14 @@
...
   <ItemGroup>
     <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="11.0.0" />
+    <PackageReference Include="AWSSDK.S3" Version="3.7.9.33" />
     <PackageReference Include="BCrypt.Net-Core" Version="1.6.0" />
     <PackageReference Include="FluentValidation" Version="10.4.0" />
     <PackageReference Include="Hangfire.Core" Version="1.7.28" />
     <PackageReference Include="Hangfire.Mongo" Version="1.7.0" />
     <PackageReference Include="Hangfire.PostgreSql" Version="1.9.7" />
     <PackageReference Include="MailKit" Version="3.2.0" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Features" Version="5.0.17" />
```
Comment (yarmolovichalex) [inline services/api-dotnet/src/app/Common/Common.csproj:18]: Where it's used?

**Review by yarmolovichalex — CHANGES_REQUESTED** [review]

File: services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs (hunk 1 above)
Comment (amonich) [reply services/api-dotnet/src/app/Api.NoSql/Controllers/UsersController.cs:80]: Filename generation logic copied from node.js version

File: services/api-dotnet/src/app/Common/Common.csproj (hunk 7 above)
Comment (amonich) [reply services/api-dotnet/src/app/Common/Common.csproj:18]: The package contains IFormFile interface, which is used in [UploadPhotoModel.cs](https://github.com/paralect/ship/blob/fdd954323d89439dc78e1d009591f2c30f8b339e/services/api-dotnet/src/app/Common/Models/View/User/UploadPhotoModel.cs)

**Review by yarmolovichalex — APPROVED** [review]

**Pull request merged**